webapp/src/components/AssetPage/BestBuyingOption/BestBuyingOption.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useEffect, useState } from 'react'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import {
+  Bid,
+  BidSortBy,
+  ListingStatus,
+  Network,
+  Order,
+  OrderFilters,
+  OrderSortBy,
+  Rarity
+} from '@dcl/schemas'
+import { Button, Loader, Mana } from 'decentraland-ui'
+import { ethers } from 'ethers'
+
+import { formatWeiMANA } from '../../../lib/mana'
+import { formatDistanceToNow } from '../../../lib/date'
+import { locations } from '../../../modules/routing/locations'
+import { isNFT } from '../../../modules/asset/utils'
+import { bidAPI, orderAPI } from '../../../modules/vendor/decentraland'
+import { ManaToFiat } from '../../ManaToFiat'
+import { LinkedProfile } from '../../LinkedProfile'
+import { BuyNFTButtons } from '../SaleActionBox/BuyNFTButtons'
+import { BuyOptions, Props } from './BestBuyingOption.types'
+import styles from './BestBuyingOption.module.css'
+
+const BestBuyingOption = ({ asset }: Props) => {
+  const [buyOption, setBuyOption] = useState<BuyOptions | null>(null)
+  const [isLoading, setIsLoading] = useState(false)
+  const [listing, setListing] = useState<{
+    order: Order
+    total: number
+  } | null>(null)
+  const [bid, setBid] = useState<Bid | null>(null)
+
+  useEffect(() => {
+    if (asset && !isNFT(asset)) {
+      if (asset.available > 1) {
+        setBuyOption(BuyOptions.MINT)
+      } else {
+        setIsLoading(true)
+
+        let params: OrderFilters = {
+          contractAddress: asset.contractAddress,
+          first: 1,
+          skip: 0,
+          status: ListingStatus.OPEN
+        }
+        const sortBy = OrderSortBy.CHEAPEST
+
+        if (asset.network === Network.MATIC && asset.itemId) {
+          params.itemId = asset.itemId
+        } else if (asset.network === Network.ETHEREUM) {
+          params.nftName = asset.name
+        }
+
+        orderAPI
+          .fetchOrders(params, sortBy)
+          .then(response => {
+            if (response.data.length > 0) {
+              setBuyOption(BuyOptions.BUY_LISTING)
+              setListing({ order: response.data[0], total: response.total })
+              bidAPI
+                .fetchByNFT(
+                  asset.contractAddress,
+                  response.data[0].tokenId,
+                  ListingStatus.OPEN,
+                  BidSortBy.MOST_EXPENSIVE,
```

Comment:
```suggestion
                  BidSortBy.MOST_EXPENSIVE,
```
I believe this is going to return us the most expensive one, right? We want the cheapest. Can we add a `orderDirection`?